Collector/ProfileFactory.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+class ProfileFactory
```

Comment:
I will try this. It sould work as I can mock the `Collector::getCurrentStack` method to return a real `Stack` object and I will be able to write some assert around it. Looks promising. If doing so, I will remove the two factories as I created both while writing tests.